File: z-clients/python/diom/apis/health.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    PingOut,
    ReadyOut,
)


class HealthAsync(ApiBase):
    async def ping(
        self,
    ) -> PingOut:
        """Verify the server is up and running.

        This endpoint only checks the server itself, not the cluster mechanism, and should not be used
        as a readiness gate."""

        return await self._request_asyncio(
            method="get",
            path="/api/v1.health.ping",
            response_type=PingOut,
        )

    async def ready(
        self,
    ) -> ReadyOut:
        """Verify that this server is ready to serve customer traffic."""

        return await self._request_asyncio(
            method="get",
            path="/api/v1.health.ready",
            response_type=ReadyOut,
        )

    async def error(
        self,
    ) -> None:
        """Intentionally return an error"""

        await self._request_asyncio_no_response(
            method="post",
            path="/api/v1.health.error",
        )


class Health(ApiBase):
    def ping(
        self,
    ) -> PingOut:
        """Verify the server is up and running.

        This endpoint only checks the server itself, not the cluster mechanism, and should not be used
        as a readiness gate."""

        return self._request_sync(
            method="get",
            path="/api/v1.health.ping",
            response_type=PingOut,
        )

    def ready(
        self,
    ) -> ReadyOut:
        """Verify that this server is ready to serve customer traffic."""

        return self._request_sync(
            method="get",
            path="/api/v1.health.ready",
            response_type=ReadyOut,
        )

    def error(
        self,
    ) -> None:
        """Intentionally return an error"""

        self._request_sync_no_response(
            method="post",
            path="/api/v1.health.error",
        )

File: z-clients/java/src/main/java/com/svix/diom/apis/Health.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.PingOut;
import com.svix.diom.models.ReadyOut;

public class Health {
    private final HttpClient client;

    public Health(HttpClient client) {
        this.client = client;
    }

    /**
* Verify the server is up and running.
* 
* This endpoint only checks the server itself, not the cluster mechanism, and should not be used
* as a readiness gate.
*/
    public PingOut ping(
        
    ) throws DiomException {

        return this.client.executeRequest(
            "GET",
            "/api/v1.health.ping",
            null,
            null,
            PingOut.class
        );
    }

    /** Verify that this server is ready to serve customer traffic. */
    public ReadyOut ready(
        
    ) throws DiomException {

        return this.client.executeRequest(
            "GET",
            "/api/v1.health.ready",
            null,
            null,
            ReadyOut.class
        );
    }

    /** Intentionally return an error */
    public void error(
        
    ) throws DiomException {

        this.client.executeRequest(
            "POST",
            "/api/v1.health.error",
            null,
            null,
            null
        );
    }
}